File: arduino_workspace/TG_v2/runMotors.cpp
#include "runMotors.h"
#include "encoders.h"
#include "pinOutIn.h"

#include <math.h>
#include <Arduino.h>
#include <util/atomic.h>

// pins A11, A12, A13, B3, B4

byte driverMode[] = {LOW, LOW, LOW};

void stepResolution(int stepResolution){
    stepResolutionEncoder(stepResolution, driverMode);

    digitalWrite(M0, driverMode[0]);
    digitalWrite(M1, driverMode[1]);
    digitalWrite(M2, driverMode[2]);
}

void convertRpmToDelays(
    int *w1_delay,
    int *w2_delay,
    int *w3_delay,
    float w1,
    float w2,
    float w3,
    int stepResolution)
{

    *w1_delay = delayMicrosecondsForStep(stepResolution, w1);
    *w2_delay = delayMicrosecondsForStep(stepResolution, w2);
    *w3_delay = delayMicrosecondsForStep(stepResolution, w3);

}


void SetSleep(int motor, byte value){

    switch (motor) {
        case 1:
            digitalWrite(SLEEP_1, value);
            break;
        case 2:
            digitalWrite(SLEEP_2, value);
            break;
        case 3:
            digitalWrite(SLEEP_3, value);
            break; 
    }

}

void SetReset(int motor, byte value){

    switch (motor) {
        case 1:
            digitalWrite(RESET_1, value);
            break;
        case 2:
            digitalWrite(RESET_2, value);
            break;
        case 3:
            digitalWrite(RESET_3, value);
            break; 
    }

}

void SetStep(int motor, byte value){

    switch (motor) {
        case 1:
            digitalWrite(STEP_1, value);
            break;
        case 2:
            digitalWrite(STEP_2, value);
            break;
        case 3:
            digitalWrite(STEP_3, value);
            break; 
    }

}

File: raspberry/TG/run_motors.py
#include "runMotors.h"
import encoders
import pin_out_in

#include <math.h>
#include <Arduino.h>
#include <util/atomic.h>


class motorSetup:

    def __init__(self, GPIO):
        self.GPIO = GPIO


    def step_resolution_output(self, step_resolution):
        driver_mode = encoders.step_resolution_encoder(step_resolution, self.GPIO)

        self.GPIO.output(pin_out_in.M0, driver_mode[0])
        self.GPIO.output(pin_out_in.M1, driver_mode[1])
        self.GPIO.output(pin_out_in.M2, driver_mode[2])


    def set_motor_sleep(self, motor, value):
        if motor==1:
            self.GPIO.output(pin_out_in.SLEEP_1, value)
        elif motor==2:
            self.GPIO.output(pin_out_in.SLEEP_2, value)
        elif motor==3:
            self.GPIO.output(pin_out_in.SLEEP_3, value)



    def set_motor_reset(self, motor, value):
        if motor==1:
            self.GPIO.output(pin_out_in.RESET_1, value)
        elif motor==2:
            self.GPIO.output(pin_out_in.RESET_2, value)
        elif motor==3:
            self.GPIO.output(pin_out_in.RESET_3, value)
